packages/qgrid-ngx/src/lib/row/trh-core.directive.ts

```diff
@@ -28,10 +29,10 @@ export class TrhCoreDirective implements DomTr, OnInit, OnDestroy {
   }
 
   ngOnInit() {
-    this.plugin.table.box.bag[this.source].addRow(this);
+    this.plugin.table.box.bag[this.source as keyof { body: Bag; head: Bag; foot: Bag }].addRow(this);
```

Comment:
keyof typeof bag?

Reply:
Bag does not contain said fields, so it's not possible to use keyof typeof, unless we add the said fields into the exported Bag class.
<img width="373" alt="image" src="https://user-images.githubusercontent.com/31434462/170999868-8ac969df-75af-4f8c-8c5c-1d55c01c2747.png">